examples/test-exec-doc.yml

```diff
@@ -0,0 +1,221 @@
+
+          # Determine the default branch (e.g., main)
+          if git show-ref --verify --quiet refs/remotes/origin/main; then
+            DEFAULT_BRANCH="main"
+          elif git show-ref --verify --quiet refs/remotes/origin/master; then
+            DEFAULT_BRANCH="master"
+          else
+            echo "Default branch is not 'main' or 'master'. Getting all Markdown files."
+            CHANGED_MARKDOWN_FILES=$(git ls-files '*.md')
+            echo "Markdown files to test:"
+            echo "$CHANGED_MARKDOWN_FILES"
+            echo "changed=true" >> $GITHUB_OUTPUT
+            echo "changed_files<<EOF" >> $GITHUB_OUTPUT
+            echo "$CHANGED_MARKDOWN_FILES" >> $GITHUB_OUTPUT
+            echo "EOF" >> $GITHUB_OUTPUT
+            exit 0
+          fi
+          echo "Default branch is $DEFAULT_BRANCH"
+          
+          # Get the last commit SHA of the default branch
+          UPSTREAM_SHA=$(git rev-parse origin/$DEFAULT_BRANCH)
+          echo "Upstream SHA: $UPSTREAM_SHA"
+          
+          # List changed Markdown files by comparing with the upstream branch
+          CHANGED_MARKDOWN_FILES=$(git diff --name-only $UPSTREAM_SHA ${{ github.sha }} | grep -E '\.md$' || true)
+
+        else
+          # Compare the latest commit with the previous one
+          CHANGED_FILES=$(git diff --name-only ${{ github.event.before }} ${{ github.sha }})
+          echo "Changed files: $CHANGED_FILES"
+          # Filter changed Markdown files
+          CHANGED_MARKDOWN_FILES=$(echo "$CHANGED_FILES" | grep -E '\.md$' || true)
+        fi
+
+        echo "Markdown files to test:"
+        echo "$CHANGED_MARKDOWN_FILES"
+
+        if [ ! -z "$CHANGED_MARKDOWN_FILES" ]; then
+          echo "changed=true" >> $GITHUB_OUTPUT
+          # Set multiline output using the '<<EOF' syntax
+          echo "changed_files<<EOF" >> $GITHUB_OUTPUT
+          echo "$CHANGED_MARKDOWN_FILES" >> $GITHUB_OUTPUT
+          echo "EOF" >> $GITHUB_OUTPUT
+        else
+          echo "changed=false" >> $GITHUB_OUTPUT
+          echo "No Markdown files changed. Exiting."
+          exit 0
+        fi
+        
+    - name: Azure CLI Login
+      if: steps.check_changes.outputs.changed == 'true'
+      uses: azure/login@v2
+      with:
+        client-id: ${{ secrets.AZURE_CLIENT_ID }}             # Azure Client ID from secrets
+        tenant-id: ${{ secrets.AZURE_TENANT_ID }}             # Azure Tenant ID from secrets
+        subscription-id: ${{ secrets.AZURE_SUBSCRIPTION_ID }} # Azure Subscription ID from secrets
+
+    - name: Set up Python
+      if: steps.check_changes.outputs.changed == 'true'
+      uses: actions/setup-python@v2
+      with:
+        python-version: '3.x'  # Specify Python version 3.x
+
+    - name: Install dependencies
+      if: steps.check_changes.outputs.changed == 'true'
+      run: |
+        pip install --upgrade pip==24.3.1 || pip install --upgrade pip  # Try pinned version, fallback to latest
```

Comment:
I don't think you need to add fall backs here; It's highly unlikely that the pinned versions will fail installation because once they're published, they're effectively available for forever

Reply:
Same for the line below

Reply:
what do you suggest? I put that because I was seeing some errors in installing those versions in the GH action shell and it seemed to fall back on the latest version. But maybe I was looking at the wrong thing. Just for posterity I added it

Reply:
What was the issue with using a pinned version? Because my suggestion is that you just use the pinned version and don't define a fallback. Any version of a python package published to pypi is immutable and it cannot be modified/taken down even if the publisher wanted to.

Reply:
done